packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -181,12 +181,10 @@ export class ExploreCompiler {
 
                             const isTimeIntervalBaseDimensionVisible =
                                 dimension.timeInterval &&
-                                dimension.groups &&
+                                dimension.timeIntervalBaseDimensionName &&
                                 join.fields
                                     ? join.fields.includes(
-                                          dimension.groups[
-                                              dimension.groups.length - 1
-                                          ],
```

Comment:
The problem with this logic is that the time interval dimension doesn't store the base dimension name in the `groups` but it's label. So this was never a match.